akka-stream-tests/src/test/scala/akka/stream/io/FileSinkSpec.scala

```diff
@@ -52,6 +53,17 @@ class FileSinkSpec extends AkkaSpec(UnboundedMailboxConfig) {
       }
     }
 
+    "create new file if not exists" in assertAllStagesStopped {
+      targetFile({ f ⇒
+        val completion = Source(TestByteStrings)
+          .runWith(FileIO.toFile(f))
+
+        val result = Await.result(completion, 3.seconds)
+        result.count should equal(6006)
+        checkFileContents(f, TestLines.mkString(""))
+      }, create = false)
+    }
+
```

Comment:
Test cases for `Set(StandardOpenOption.APPEND)` and `Set(StandardOpenOption.WRITE)` with `create = false` would be nice.